pkg/cmd/roachtest/spec/cluster_spec.go

```diff
@@ -153,12 +154,30 @@ func MakeClusterSpec(nodeCount int, opts ...Option) ClusterSpec {
 
 // ClustersCompatible returns true if the clusters are compatible, i.e. the test
 // asking for s2 can reuse s1.
-func ClustersCompatible(s1, s2 ClusterSpec) bool {
-	s1.Lifetime = 0
-	s2.Lifetime = 0
+func ClustersCompatible(s1, s2 ClusterSpec, cloud string) bool {
+	// only consider the specification of the cloud that we are running in
+	clearClusterSpecFields(&s1, cloud)
+	clearClusterSpecFields(&s2, cloud)
 	return s1 == s2
 }
 
+func clearClusterSpecFields(cs *ClusterSpec, cloud string) {
```

Comment:
May be good to add a comment to this function explaining that it will clear everything except the passed cloud. The name of the function makes it sound a bit like it will clear the passed cloud, so a comment will make it clearer what's happening.